INSTALL.rst

```diff
@@ -44,6 +44,18 @@ From source
   git clone https://github.com/k2-fsa/k2.git
   cd k2
   python3 setup.py install
+  
+From source (with CUDA support)
+=========================
+
+.. code-block:: bash
+
+  git clone https://github.com/k2-fsa/k2.git
+  cd k2
+  export K2_CMAKE_ARGS="-DK2_WITH_CUDA=ON -DCMAKE_BUILD_TYPE=Release"
```

Comment:
Can you build k2 with CUDA support without specifying these lines explicitly?

Reply:
No. I tried without it and arrived to the system that doesn't use GPU:

~$ python3 -m k2.version
Collecting environment information...

k2 version: 1.13
Build type: Release
Git SHA1: 47c4b754bb418b2a40c3ee0f24ca5ed12b08997f
Git date: Sat Jan 29 09:39:32 2022
Cuda used to build k2: 11.3
cuDNN used to build k2: 8.2.0
Python version used to build k2: 3.6
OS used to build k2: Ubuntu 18.04.6 LTS
CMake version: 3.10.2
GCC version: 7.5.0
CMAKE_CUDA_FLAGS:
CMAKE_CXX_FLAGS:  -D_GLIBCXX_USE_CXX11_ABI=0 -Wno-unused-variable -Wno-strict-overflow
PyTorch version used to build k2: 1.10.2+cu113
PyTorch is using Cuda:
NVTX enabled: False
With CUDA: False
Disable debug: True
Sync kernels : False
Disable checks: False

However, when I mention those exports, then I get the following system:

k2 version: 1.13
Build type: Release
Git SHA1: 47c4b754bb418b2a40c3ee0f24ca5ed12b08997f
Git date: Sat Jan 29 09:39:32 2022
Cuda used to build k2: 11.3
cuDNN used to build k2: 8.2.0
Python version used to build k2: 3.6
OS used to build k2: Ubuntu 18.04.6 LTS
CMake version: 3.10.2
GCC version: 7.5.0
CMAKE_CUDA_FLAGS:  --expt-extended-lambda -gencode arch=compute_35,code=sm_35 --expt-extended-lambda -gencode arch=compute_50,code=sm_50 --expt-extended-lambda -gencode arch=compute_60,code=sm_60 --expt-extended-lambda -gencode arch=compute_61,code=sm_61 --expt-extended-lambda -gencode arch=compute_70,code=sm_70 --expt-extended-lambda -gencode arch=compute_75,code=sm_75 --expt-extended-lambda -gencode arch=compute_80,code=sm_80 --expt-extended-lambda -gencode arch=compute_86,code=sm_86 -D_GLIBCXX_USE_CXX11_ABI=0 --compiler-options -Wall --compiler-options -Wno-unknown-pragmas --compiler-options -Wno-strict-overflow
CMAKE_CXX_FLAGS:  -D_GLIBCXX_USE_CXX11_ABI=0 -Wno-strict-overflow
PyTorch version used to build k2: 1.10.2+cu113
PyTorch is using Cuda: 11.3
NVTX enabled: True
With CUDA: True
Disable debug: True
Sync kernels : False
Disable checks: False

+++

BTW, I haven't mentioned "-DCMAKE_BUILD_TYPE=Release", but arrived to a Release build (if I interpret "Disable debug: True" correctly). It looks like it is the "Release" config that is default. But explicitly mentioning  "DCMAKE_BUILD_TYPE=Release" allows to avoid any kind of ambiguity.

Reply:
> No. I tried without it and arrived to the system that doesn't use GPU:

Ok, it detects whether NVCC is available. If so, it will build a CUDA version, otherwise a CPU version is built.

---

> BTW, I haven't mentioned "-DCMAKE_BUILD_TYPE=Release", but arrived to a Release build (if I interpret "Disable debug: True" correctly)

It is
```
Build type: Release
```
that indicates the type of the build.  If you don't specify `K2_CMAKE_ARGS`, then it defaults to a Release build.
But if you specify it and don't provide `-DCMAKE_BUILD_TYPE=Release`, then it will be a Debug build.

Reply:
>that indicates the type of the build. If you don't specify K2_CMAKE_ARGS, then it defaults to a Release build.
But if you specify it and don't provide -DCMAKE_BUILD_TYPE=Release, then it will be a Debug build.

OK, got it! Thank you for the explanation.